File: CodeEval/easy/039happynumbers/solution.py
#!/usr/bin/env python

from sys import argv

def happy_proc(n):
    sum = 0
    for i in str(n):
        sum += int(i) ** 2
    return sum

for line in open(argv[1]):
    limit = 30
    n = int(line)
    while (n != 1 and limit > 0):
        n = happy_proc(n)
        limit -= 1
    print("1" if n == 1 else "0")

File: CodeEval/easy/039happynumbers/solution.rb
#!/usr/bin/env ruby

def happy_proc(n)
  n.to_s.split("").reduce(0) { |sum, x| sum + x.to_i**2 }
end

File.open(ARGV[0]).each_line do |line|
  max = 30
  n = line.to_i
  until n == 1 or max <= 0
    n = happy_proc(n)
    max -= 1
  end
  puts n == 1 ? "1" : "0"
end
